packages/shared/src/components/drawers/Drawer.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { ReactElement, ReactNode, useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { RootPortal } from '../tooltips/Portal';
+import useDebounce from '../../hooks/useDebounce';
+
+export enum DrawerPosition {
+  Bottom = 'bottom',
+  Top = 'top',
+}
+
+interface ClassName {
+  overlay?: string;
+  drawer?: string;
+}
+
+export interface DrawerProps {
+  children: ReactNode;
+  className?: ClassName;
+  position?: DrawerPosition;
+  closeOnOutsideClick?: boolean;
+  onClose(): void;
+}
+
+const drawerPositionToClassName: Record<DrawerPosition, string> = {
+  [DrawerPosition.Bottom]: 'bottom-0 rounded-t-16',
+  [DrawerPosition.Top]: 'top-0 rounded-b-16',
+};
+
+const animatePositionClassName: Record<DrawerPosition, string> = {
+  [DrawerPosition.Bottom]: 'translate-y-full',
+  [DrawerPosition.Top]: '-translate-y-full',
+};
+
+export function Drawer({
+  children,
+  className = {},
+  position = DrawerPosition.Bottom,
+  closeOnOutsideClick = true,
+  onClose,
+}: DrawerProps): ReactElement {
+  const [hasAnimated, setHasAnimated] = useState(false);
+  const [animate] = useDebounce(() => setHasAnimated(true), 1);
```

Comment:
Why do we need the debounce here?

Reply:
Same here https://github.com/dailydotdev/apps/pull/2666#discussion_r1497149472